### ISSUE TITLE
Docker dies 

### ISSUE THREAD

**Issue by nfrik** [gw0/docker-keras-full#5]
No matter what I try, docker dies instantly. Runing under ubuntu 16.04.

After running:
docker run -d -it --name kerasfull -p=6006:6006 -p=8889:8888 -v=/srv/notebooks:/srv gw000/keras-full

Exit status (1)

With logs:
[I 23:51:43.417 NotebookApp] Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
[W 23:51:43.954 NotebookApp] WARNING: The notebook server is listening on all IP addresses and not using encryption. This is not recommended.
[C 23:51:43.965 NotebookApp] Running as root is not recommended. Use --allow-root to bypass.

**Below is the output from docker inspect:**
docker inspect kerasfull 
[
    {
        "Id": "3650638867e1d04549f7bda8c6223eca6984a81a9edf4cf8abc12709077ca5fe",
        "Created": "2018-01-12T23:54:01.919628654Z",
        "Path": "/bin/sh",
        "Args": [
            "-c",
            "/bin/bash -c 'jupyter notebook --no-browser --ip=* --NotebookApp.password=\"$PASSWD\" \"$@\"'"
        ],
        "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2018-01-12T23:54:02.43346543Z",
            "FinishedAt": "2018-01-12T23:54:03.254929863Z"
        },
        "Image": "sha256:7727d5d86ee6f46c4a1f911aea815a209898e9e38f42b86a3a340fd5cbef3a94",
        "ResolvConfPath": "/var/lib/docker/containers/3650638867e1d04549f7bda8c6223eca6984a81a9edf4cf8abc12709077ca5fe/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/3650638867e1d04549f7bda8c6223eca6984a81a9edf4cf8abc12709077ca5fe/hostname",
        "HostsPath": "/var/lib/docker/containers/3650638867e1d04549f7bda8c6223eca6984a81a9edf4cf8abc12709077ca5fe/hosts",
        "LogPath": "/var/lib/docker/containers/3650638867e1d04549f7bda8c6223eca6984a81a9edf4cf8abc12709077ca5fe/3650638867e1d04549f7bda8c6223eca6984a81a9edf4cf8abc12709077ca5fe-json.log",
        "Name": "/kerasfull",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/srv/notebooks:/srv"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "6006/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "6006"
                    }
                ],
                "8888/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "8889"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "shareable",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/8eef135fbb75d67d2519307a7f0526b2617520c0ba37eedcb450928d70912371-init/diff:/var/lib/docker/overlay2/cae8808e8c1d3faddf5f02f2c6042d8808d80d3c135f97b85c1261d8d8f2ddeb/diff:/var/lib/docker/overlay2/bfb57f9266d4f52a9f573ea063618d0ea1677e268cb527b6c1b10dccba95e568/diff:/var/lib/docker/overlay2/f6ee40861289c33c8d84dcac0f7849fa5f1993488eda0116bb730175cc624b79/diff:/var/lib/docker/overlay2/4c48e0104dc046772ef12e95ba22d07f0300f3cf12867ea967c39f29a1d5a43a/diff:/var/lib/docker/overlay2/cd452ad82647e90dbb808c10bb5eccec11f6b9449fe8332333f9f5162f5dce43/diff:/var/lib/docker/overlay2/7e8f14318e9f609103724b0172b325afe9318357bede28cb6c8e0965c520cc0b/diff:/var/lib/docker/overlay2/8a2557198495486edc251ce7109e9284ea2e99bba12809ec8a8c9fc9d2018be3/diff:/var/lib/docker/overlay2/070cde4afd4b6e6829debbf2445e2a56689cd9dbf83621e7315ff9ad26976abd/diff:/var/lib/docker/overlay2/19f9db1cfec44d01f5b2bfc1834915402ad0b479aafc0d484a32b0f810c582dc/diff:/var/lib/docker/overlay2/aaec5ebcd7ebf3ab6c7182ee42d8d1ee6fbcf196272cb4a5ece125f8b9f638cc/diff:/var/lib/docker/overlay2/5aeaa3997918a8cbd25e9275aff71f892230cba185f92e0151f928907714aba6/diff:/var/lib/docker/overlay2/16002e582c08a271da8398677bb29a2c68f8c6a37e74abc6332401d649e523c4/diff:/var/lib/docker/overlay2/f9af96b8a4b31030a768d25ac67fe07e01166c13247c94f6fbf1babbe9957e5c/diff:/var/lib/docker/overlay2/9cb5f59913da5900e662225db26fff14e48b64606075e1bd89f8308282a4a4ba/diff:/var/lib/docker/overlay2/83d6d02ad1ad2cb8d35280a4ca3689a3560cd4b390d9cd3bd23cea707260c672/diff:/var/lib/docker/overlay2/3f3d9acb91302de862519419c285a0af582cfc3fb9a9c025a874e3ca65b620f8/diff:/var/lib/docker/overlay2/26716c8a1588c8d67f5c6370aed447df572a1cb89b196390ae88276b228b8d6f/diff:/var/lib/docker/overlay2/c5f9c9978d5dde130def9109bd95526bb93cc4846e8e3580e70c7c069e791ecb/diff:/var/lib/docker/overlay2/ff498ff84a772f6b98ddec7f5a9203f2657545400d7ca73b016b52603359d861/diff:/var/lib/docker/overlay2/60288f1728a3f175924117fe267b804fe3f16bf94fa5957b257416c0a7d5a832/diff:/var/lib/docker/overlay2/e4c4911eca518489880902e8da62e63ec6894d4a679ef56f881333b235adebbb/diff:/var/lib/docker/overlay2/ec6360262c4cdfb60728e7311fc242bb3474379d89781bc5232ba2b8e23a4dfa/diff",
                "MergedDir": "/var/lib/docker/overlay2/8eef135fbb75d67d2519307a7f0526b2617520c0ba37eedcb450928d70912371/merged",
                "UpperDir": "/var/lib/docker/overlay2/8eef135fbb75d67d2519307a7f0526b2617520c0ba37eedcb450928d70912371/diff",
                "WorkDir": "/var/lib/docker/overlay2/8eef135fbb75d67d2519307a7f0526b2617520c0ba37eedcb450928d70912371/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/srv/notebooks",
                "Destination": "/srv",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "3650638867e1",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "6006/tcp": {},
                "8888/tcp": {}
            },
            "Tty": true,
            "OpenStdin": true,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "DEBIAN_FRONTEND=noninteractive",
                "KERAS_BACKEND=tensorflow",
                "THEANO_FLAGS=device=cpu,floatX=float32",
                "PASSWD=sha1:98b767asd:8daasdfasdfasdfasdfasdf69edc97737asdf407824"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "/bin/bash -c 'jupyter notebook --no-browser --ip=* --NotebookApp.password=\"$PASSWD\" \"$@\"'"
            ],
            "ArgsEscaped": true,
            "Image": "gw000/keras-full",
            "Volumes": null,
            "WorkingDir": "/srv",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {}
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "8fc0e32b83fcf67f3d7ae7addc21c7e0858954c888a969c18372ff580050cc05",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/8fc0e32b83fc",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "86342c838fe1853d899c92a3ca14f19f3a64fc520363715cfda27c678a8fcda5",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]

**Comment by gw0** [comment]
You are right, the newest version of Jupyter produces an error if you run it as root, but to no provide the `--allow-root` parameter (`Running as root is not recommended. Use --allow-root to bypass.`).

It will be fixed in the next version, as a workaround you can start it with:

```
sudo docker run -d -it --name kerasfull -p=6006:6006 -p=8889:8888 gw000/keras-full /bin/bash -c 'jupyter notebook --no-browser --allow-root --ip=* --NotebookApp.password="sha1:98b767162d34:8da1bc3c75a0f29145769edc977375a373407824"'
```
